src/ui/components/Library/useFilters.ts

```diff
@@ -0,0 +1,103 @@
+import { createContext, useState } from "react";
+import { Recording } from "ui/types";
+import { getParams, updateUrlWithParams } from "ui/utils/environment";
+
+export type LibraryFilters = {
+  searchString: string;
+  qualifiers: {
+    created?: string;
+    target?: string;
+  };
+};
+
+export const LibraryFiltersContext = createContext({
+  searchString: "",
+  qualifiers: {},
+});
+
+function parseFilterString(str: string): LibraryFilters {
+  const words = str.split(" ");
+
+  // If there are multiple matching entries for the same filter, use the last one.
+  // TODO: Add validation for created and target. It should return null if the user didn't comply with
+  // the expected format.
+  const created = words.filter(w => !!w.match(/^created:/)).pop();
+  const target = words.filter(w => !!w.match(/^target:/)).pop();
+  const searchString = words.filter(w => !w.match(/created:|target:/)).join(" ");
+
+  return { qualifiers: { created, target }, searchString };
+}
+
+const subStringInString = (subString: string, string: string | null) => {
+  if (!string) {
+    return false;
+  }
+
+  return string.toLowerCase().includes(subString.toLowerCase());
+};
+
+export const filterRecordings = (recordings: Recording[], filters: LibraryFilters) => {
+  const { searchString, qualifiers } = filters;
+  let filteredRecordings = recordings;
+
+  filteredRecordings = searchString
+    ? recordings.filter(
+        r => subStringInString(searchString, r.url) || subStringInString(searchString, r.title)
+      )
+    : recordings;
+  filteredRecordings =
+    qualifiers.target && qualifiers.target === "target:node"
+      ? filteredRecordings.filter(r => !r.user)
+      : filteredRecordings;
+  filteredRecordings =
+    qualifiers.created && new Date(qualifiers.created)
+      ? filteredRecordings.filter(
+          r => new Date(r.date).getTime() > new Date(qualifiers.created!).getTime()
+        )
+      : filteredRecordings;
+
+  return filteredRecordings;
+};
+
+const useFilterString = (str: string) => {
+  const [appliedString, setAppliedString] = useState(str);
+  const [displayedString, setDisplayedString] = useState(str);
+
+  const applyDisplayedString = () => {
+    setAppliedString(displayedString);
+    updateUrlWithParams({ q: displayedString });
+  };
+  const forceSetAppliedString = (newStr: string) => {
+    setAppliedString(newStr);
+    setDisplayedString(newStr);
+    updateUrlWithParams({ q: newStr });
+  };
```

Comment:
The naming in this hook is confusing to me.

There's a "display string" and an "applied string" but then there's a method called "apply displayed string" 🙃 

Also `forceSetAppliedString` updates both applied and displayed strings, but `applyDisplayedString` updates only _applied_ string which is the... opposite of what I'd expect 😆 

Can we change these names to be clearer?

I think `displayedString` and `appliedString` make sense 👍🏼  but the callbacks could be made clearer? Seems like there are 3 use-cases:
1. Update the displayed string only (as a user types).
2. Update the displayed and applied strings (choosing a pre-set filter from the dropdown).
3. Applying the current display string (as a user presses "Enter" after updating the filter).

I think we could simplify to the following two functions:
* `setDisplayText` (alias to state updater) – handles case 1
* `setAppliedText` (updates both displayed and applied texts) – handles cases 2 and 3